magicctapipe/io/io.py

```diff
@@ -1032,24 +876,36 @@ def load_dl2_data_file(input_file, quality_cuts, event_type, weight_type_dl2):
         If the input event type is not known
     """
 
+    TEL_NAMES, TEL_COMBINATIONS = telescope_combinations(config)
+    combo_types = np.asarray(range(len(TEL_COMBINATIONS)))
+    three_or_more = []
+    for n, combination in enumerate(TEL_COMBINATIONS.values()):
+        if len(combination) >= 3:
+            three_or_more.append(n)
+
     # Load the input file
     event_data = pd.read_hdf(input_file, key="events/parameters")
     event_data.set_index(["obs_id", "event_id", "tel_id"], inplace=True)
     event_data.sort_index(inplace=True)
 
-    event_data = get_stereo_events_old(event_data, quality_cuts)
+    event_data = get_stereo_events(
+        event_data, config, quality_cuts, eval_multi_combo=False
+    )
 
     logger.info(f"\nExtracting the events of the '{event_type}' type...")
 
     if event_type == "software":
         # The events of the MAGIC-stereo combination are excluded
-        event_data.query("combo_type > 0", inplace=True)
+        event_data.query(f"combo_type < {combo_types[-1]}", inplace=True)
+
+    elif event_type == "software_3tels_or_more":
+        event_data.query(f"combo_type == {three_or_more}", inplace=True)
 
-    elif event_type == "software_only_3tel":
-        event_data.query("combo_type == 3", inplace=True)
+    elif event_type == "software_6_tel":
+        event_data.query(f"combo_type < {combo_types[-1]}", inplace=True)
 
     elif event_type == "magic_only":
-        event_data.query("combo_type == 0", inplace=True)
+        event_data.query(f"combo_type == {combo_types[-1]}", inplace=True)
 
     elif event_type == "hardware":
         logger.warning(
```

Comment:
hardware trigger is already implemented in ctapipe_io_magic

Reply:
fixed